### DISCUSSION TITLE
Table edit and view actions not working as expected

### DISCUSSION THREAD

**Discussion by Jam0r85** [filamentphp/filament#8887] [Help]
I don't think this is a package issue but more that i've got something setup wrong and I cannot figure it out.

When using a resource table inside a livewire component the table view and edit actions don't work. The view action does nothing and the edit action, when clicked, just shows a 'Saved' notification.

When using the resource directly it works fine so it must be an issue with using it inside a livewire component.

The livewire component is as follows

```
<?php

namespace App\Livewire;

use App\Filament\Resources\GasTaskResource;
use App\Models\GasTask;
use Filament\Forms\Concerns\InteractsWithForms;
use Filament\Forms\Contracts\HasForms;
use Filament\Tables\Concerns\InteractsWithTable;
use Filament\Tables\Contracts\HasTable;
use Filament\Tables\Table;
use Illuminate\View\View;
use Livewire\Component;

class GasTasksTable extends Component implements HasForms, HasTable
{
    use InteractsWithForms;
    use InteractsWithTable;

    public function table(Table $table): Table
    {
        return GasTaskResource::table($table)
            ->heading('Jobs')
            ->query(GasTask::query());
    }

    public function render(): View
    {
        return view('livewire.gas-tasks-table');
    }
}

```

Any ideas?

**Comment by VincvanAs** [comment]
There is a page in the filament documentation completely dedicated to adding a filament table to a livewire component: https://filamentphp.com/docs/3.x/tables/adding-a-table-to-a-livewire-component

**Comment by Jam0r85** [comment]
Which I have used. I'm trying to understand why when viewing the table inside the resource all actions work as expected, however when the same table is implemented within a livewire component, the edit action refuses to work.

There is no mention about having to adjust a table when using it inside a livewire component.

**Reply by cakanm** [reply]
I have the same question. Have you found the solution?
I assume that something is missing in livewire component and it doesn't know which forms to use. But I don't know how to pass any additional data.

**Comment by saqibapps** [comment]
Has anyone found a solution to the situation here?

**Comment by coolsam726** [comment]
I was facing the same challenge, I managed to make it work by overriding `table actions` in the Livewire Component as follows:
```php
use Filament\Forms\Form;
use Filament\Infolists\Infolist;
use Filament\Tables\Actions\DeleteAction;
use Filament\Tables\Actions\EditAction;
use Filament\Tables\Actions\ViewAction;


    public function table(Table $table): Table
    {
        return ProductTypeResource::table($table)
            ->query(ProductType::query()->where('active','=', true))
            ->actions([
                ViewAction::make()
                    ->infolist(fn(Infolist $infolist) => ProductTypeResource::infolist($infolist))
                    ->form(fn(Form $form) => ProductTypeResource::form($form)),
                EditAction::make()->form(fn(Form $form) => ProductTypeResource::form($form)),
                DeleteAction::make(),
            ]);
    }
```
I hope this will help you @Jam0r85 and @saqibapps

**Comment by saqibapps** [comment]
Hi. Thank you for your response. Its a good idea. What im finding is that using your method the Modal now loads from the view/edit action on a table. 
However, The modals all open in view mode. You can't edit them. Neither is the styling in the modal form to view/edit the record.
Any ideas?

**Reply by smiliyas** [reply]
Can you share the snap of your view/edit modal form view. It will be easier to understand the problem.

**Reply by saqibapps** [reply]
using @coolsam726 technique the action buttons now produce modals so i can see forms...but they are readonly. Its exactly the issue @Jam0r85 explained that he was having above.  Here is what the EditAction form looks like and as you can see its the view-only look:

![image](https://github.com/filamentphp/filament/assets/74057149/28d0cb17-1a89-4823-bca6-0bb5ecbd058b)

This is what it should look like:

![image](https://github.com/filamentphp/filament/assets/74057149/b182e105-7f7e-4fa4-8a61-2d176af584e6)

Many Thanks

**Reply by coolsam726** [reply]
Try the same technique on a fresh installation, maybe you missed a step in configuring forms and form actions in livewire.
